### ISSUE TITLE
NoMethodError: undefined method `register_postprocessor' for nil:NilClass

### ISSUE THREAD

**Issue by madkin10** [ai/autoprefixer-rails#157]
Getting this error running any rails/rake commands:
```
rake aborted!
NoMethodError: undefined method `register_postprocessor' for nil:NilClass
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/autoprefixer-rails-9.5.1/lib/autoprefixer-rails/sprockets.rb:32:in `install'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/autoprefixer-rails-9.5.1/lib/autoprefixer-rails.rb:20:in `install'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/autoprefixer-rails-9.5.1/lib/autoprefixer-rails/railtie.rb:18:in `block in <class:Railtie>'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/railties-5.2.3/lib/rails/initializable.rb:32:in `instance_exec'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/railties-5.2.3/lib/rails/initializable.rb:32:in `run'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/railties-5.2.3/lib/rails/initializable.rb:61:in `block in run_initializers'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/railties-5.2.3/lib/rails/initializable.rb:60:in `run_initializers'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/railties-5.2.3/lib/rails/application.rb:361:in `initialize!'
/Users/madkins/projects/business_continuity/dashboard/config/environment.rb:5:in `<top (required)>'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/railties-5.2.3/lib/rails/application.rb:337:in `require'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/railties-5.2.3/lib/rails/application.rb:337:in `require_environment!'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/railties-5.2.3/lib/rails/application.rb:520:in `block in run_tasks_blocks'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/bundler-2.0.1/lib/bundler/cli/exec.rb:74:in `load'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/bundler-2.0.1/lib/bundler/cli/exec.rb:74:in `kernel_load'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/bundler-2.0.1/lib/bundler/cli/exec.rb:28:in `run'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/bundler-2.0.1/lib/bundler/cli.rb:463:in `exec'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/bundler-2.0.1/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/bundler-2.0.1/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/bundler-2.0.1/lib/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/bundler-2.0.1/lib/bundler/cli.rb:27:in `dispatch'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/bundler-2.0.1/lib/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/bundler-2.0.1/lib/bundler/cli.rb:18:in `start'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/bundler-2.0.1/exe/bundle:30:in `block in <top (required)>'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/bundler-2.0.1/lib/bundler/friendly_errors.rb:124:in `with_friendly_errors'
/Users/madkins/.rvm/gems/ruby-2.6.1/gems/bundler-2.0.1/exe/bundle:22:in `<top (required)>'
/Users/madkins/.rvm/gems/ruby-2.6.1/bin/bundle:23:in `load'
/Users/madkins/.rvm/gems/ruby-2.6.1/bin/bundle:23:in `<main>'
Tasks: TOP => db:drop => db:load_config => environment
(See full trace by running task with --trace)
```

Bundle List:
```
  * actioncable (5.2.3)
  * actionmailer (5.2.3)
  * actionpack (5.2.3)
  * actionview (5.2.3)
  * activejob (5.2.3)
  * activemodel (5.2.3)
  * activerecord (5.2.3)
  * activestorage (5.2.3)
  * activesupport (5.2.3)
  * arel (9.0.0)
  * autoprefixer-rails (9.5.1)
  * aws-eventstream (1.0.2)
  * aws-partitions (1.149.0)
  * aws-sdk-core (3.48.3)
  * aws-sdk-kms (1.16.0)
  * aws-sdk-s3 (1.36.0)
  * aws-sigv4 (1.1.0)
  * bindex (0.7.0)
  * bootstrap (4.3.1)
  * builder (3.2.3)
  * bundler (2.0.1)
  * byebug (11.0.1)
  * concurrent-ruby (1.1.5)
  * crass (1.0.4)
  * domain_name (0.5.20180417)
  * erubi (1.8.0)
  * execjs (2.7.0)
  * ffi (1.10.0)
  * globalid (0.4.2)
  * http-cookie (1.0.3)
  * i18n (1.6.0)
  * jmespath (1.4.0)
  * jquery-rails (4.3.3)
  * listen (3.1.5)
  * loofah (2.2.3)
  * mail (2.7.1)
  * marcel (0.3.3)
  * method_source (0.9.2)
  * mime-types (3.2.2)
  * mime-types-data (3.2019.0331)
  * mimemagic (0.3.3)
  * mini_mime (1.0.1)
  * mini_portile2 (2.4.0)
  * minitest (5.11.3)
  * netrc (0.11.0)
  * nio4r (2.3.1)
  * nokogiri (1.10.2)
  * pg (0.21.0)
  * popper_js (1.14.5)
  * puma (3.12.1)
  * rack (2.0.7)
  * rack-test (1.1.0)
  * rails (5.2.3)
  * rails-dom-testing (2.0.3)
  * rails-html-sanitizer (1.0.4)
  * railties (5.2.3)
  * rake (12.3.2)
  * rb-fsevent (0.10.3)
  * rb-inotify (0.10.0)
  * rest-client (2.0.2)
  * ruby_dep (1.5.0)
  * sass (3.7.4)
  * sass-listen (4.0.0)
  * sass-rails (5.0.7)
  * sassc (2.0.1)
  * sassc-rails (2.1.0)
  * spring (2.0.2)
  * spring-watcher-listen (2.0.1)
  * sprockets (3.7.2)
  * sprockets-rails (3.2.1)
  * thor (0.20.3)
  * thread_safe (0.3.6)
  * tilt (2.0.9)
  * tzinfo (1.2.5)
  * unf (0.1.4)
  * unf_ext (0.0.7.5)
  * web-console (3.7.0)
  * websocket-driver (0.7.0)
  * websocket-extensions (0.1.3)
```
 Seems related to: https://github.com/ai/autoprefixer-rails/issues/52

**This error goes away if I lock `autoprefixer-rails` to version `9.4.9`**

**Comment by ai** [comment]
Can you update Sprockets?

**Comment by balbesina** [comment]
This is pretty common error for developers that don't understand ruby parser's operator "weight". 
See: https://github.com/ai/autoprefixer-rails/blob/master/lib/autoprefixer-rails/railtie.rb#L17
``` if defined? app.assets && !app.assets.nil? ```
Here the `defined?` method has more "weight" for parser than `&&`, that's why it is always better to encolse the defined? argument into braces, otherwise it executes like:
``` if defined?(app.assets && !app.assets.nil?) ```
but I guess the author wanted it to be like:
``` if defined?(app.assets) && !app.assets.nil? ```

That way this bug passes nil to `AutoprefixerRails.install(app.assets, config)` call and finally raises the `undefined method for NilClass'

**Comment by ai** [comment]
@balbesina wow, thank. Can you send PR?

**Comment by ai** [comment]
Fixed f1e8122

**Comment by ai** [comment]
Released in 9.5.1.1
